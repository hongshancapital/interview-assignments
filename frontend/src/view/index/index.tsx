import React, { Fragment } from "react";
import Carousel from "../../components/carousel";
const contentStyle: React.CSSProperties = {
  height: "100vw",
  color: "#fff",
  lineHeight: "100vh",
  textAlign: "center",
  background: "#364d79",
  width: "100vw",
};
const Index = () => {
  return (
    <Fragment>
      {/* <div className="index_container">6666</div> */}
      <Carousel>
        <div>
          <h3 style={contentStyle}>1</h3>
        </div>
        <div>
          <h3 style={contentStyle}>2</h3>
        </div>
        <div>
          <h3 style={contentStyle}>3</h3>
        </div>
        <div>
          <h3 style={contentStyle}>4</h3>
        </div>
      </Carousel>
    </Fragment>
  );
};
export default Index;
