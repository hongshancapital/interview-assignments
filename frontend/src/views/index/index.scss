.content {
  height: 100vh;
  width: 100vw;
  padding-top: 100px;
  text-align: center;
  position: relative;
  font-size: 24px;
  float: left;
  h1 {
    font-size: 50px;
  }
}
.content1 {
  color: #fff;
  background:url("../../assets/iphone.png") no-repeat center/75% rgb(17, 17, 17);
  background-position: center 55%;
  h1 {
    color: white !important;
  }
}
.content2 {
  background:url("../../assets/tablet.png") no-repeat center/130% rgb(250, 250, 250);
  background-position: center 70%;
}
.content3 {
  background:url("../../assets/airpods.png") no-repeat center/190% rgb(241, 241, 243);
  background-position: center 75%;
}
