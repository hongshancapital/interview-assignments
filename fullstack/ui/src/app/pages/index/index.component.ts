import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-index',
  templateUrl: './index.component.html',
  styleUrls: ['./index.component.scss']
})
export class IndexComponent implements OnInit {

  constructor() { }

  ngOnInit(): void {
  }

}
