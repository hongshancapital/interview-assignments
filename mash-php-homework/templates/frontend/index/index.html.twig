<!doctype html>
<html className="no-js" lang="en" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>用户注册</title>
       
        <style>
            #error-info {
                color: red;
            }
        </style> 
    </head>
    <body>
    <!--main-->
    <div className="grid-container" style="max-width: 100rem; margin:50px;">
        <div className="grid-x" style="min-height:581px;">                
            <div className="cell padding-left-1 margin-top-1 margin-bottom-1" id="root">                   
                 

            </div>
        </div>
    </div>

  
   
    <script src="https://unpkg.com/react@17/umd/react.production.min.js" crossorigin></script>
<script src="https://unpkg.com/react-dom@17/umd/react-dom.production.min.js" crossorigin></script>
 
    <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>  
    <script type="text/babel">
        
        const User = () => {
            const [formData, setFormData] = React.useState({
                username: '',
                password: '',
                passwordRepeat: '',
            })
            const [response, setResponse] = React.useState('')
            const [resource, setResource] = React.useState([])
          
            const url = "{{ path('api.register') }}"  
            const urlResource = "{{ path('api.users') }}"  
            
            const httpRequest = (url, data = {}) => {                
                fetch(url, {
                                method: 'POST', 
                                body: JSON.stringify(data), 
                                headers: new Headers({
                                    'Content-Type': 'application/json'
                                })
                })
                .then(res => res.json())
                .then(json => {   
                    setResponse(json.msg)
                    if (url == "{{ path('api.register') }}" && json.code == 0) {
                        let newResouce = resource
                        newResouce.push({username:formData.username})
                        setResource(newResouce)  
                        setFormData({
                            username: '',
                            password: '',
                            passwordRepeat: '',
                        })
                    } else if (url == "{{ path('api.users') }}" ) {
                         setResource(json.data) 
                    }
                })
                .catch(error => console.error('Error:', error))
            }

            React.useEffect(() => httpRequest(urlResource), [])
           
            const handleChange = (event) => {
                setFormData({
                    ...formData,
                    [event.target.id]:event.target.value,
                })
            }
            
            const handleSubmit = () => {
                httpRequest(url, formData)
            }

            const Item = () => {
                return resource.map((item, index) => {                    
                    return <li key={index}>{item.username}</li>
                })
            }

            return (
                <React.Fragment>
                    <div className="grid-x">
                        <div className="cell large-offset-2">
                            <p className="lead">用户注册</p>
                            <p id="error-info" className="help-text">{response}</p>
                        </div>
                        <div className="cell small-2">
                            <label className="text-right middle">
                            用户名:
                            </label>
                        </div>
                        <div className="cell small-10">
                            <input type="text" id="username" placeholder="用户名" value={formData.username} onChange={(event)=>handleChange(event)} />
                        </div>
                        <div className="cell small-2">
                            <label className="text-right middle">
                            初始密码:
                            </label>
                        </div>
                        <div className="cell small-10">
                            <input type="password" id="password"placeholder="初始密码" value={formData.password} onChange={(event)=>handleChange(event)} />
                        </div>
                        <div className="cell small-2">
                            <label className="text-right middle">
                            确认密码:
                            </label>
                        </div>
                        <div className="cell small-10">
                            <input type="password" id="passwordRepeat" placeholder="确认密码" value={formData.passwordRepeat} onChange={(event)=>handleChange(event)} />
                        </div>
                        
                        <div className="cell small-offset-4">
                            <button type="button" className="button small" onClick={()=>handleSubmit()}>提交</button>                
                        </div>
                    </div>
                    <div className="grid-x margin-top-3">
                        <div className="cell large-offset-2">
                            <p className="help-text">用户列表</p>
                            <ol>
                                <Item />
                            </ol>
                        </div>
                    </div>                   
                </React.Fragment>
            )
        }


        const domContainer = document.querySelector('#root');
        ReactDOM.render(<User />, domContainer);
    </script>
    </body>
</html>