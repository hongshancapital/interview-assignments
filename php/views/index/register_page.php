<!doctype html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport"
          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>用户注册页</title>
</head>
<body>
<div>
    <div>
        <label for="">Username：</label>
        <input type="text" class="UserName" id="UserName" onblur="checkUserName()">
        <span id="ver_UserNameMsg"></span>
    </div>
    <div>
        <label for="">Password：</label>
        <input type="password" class="Password" id="Password" onblur="checkPassword()">
        <span id="ver_PasswordMsg"></span>
    </div>
    <div>
        <label for="">Repeat Password：</label>
        <input type="password" class="RepeatPassword" id="RepeatPassword" onblur="checkRepeatPassword()">
        <span id="ver_RepeatPasswordMsg"></span>
    </div>

    <div>
        <button type="button" class="submit">submit</button>
    </div>
</div>
</body>
</html>
<script src="https://code.jquery.com/jquery-3.1.1.min.js"></script>
<script>
    var flagUser = false;
    var flagPwd = false;
    var flagRpwd = false;

    function checkUserName() {
        var userName = $("#UserName").val();
        var regex = /^[a-zA-Z_][a-zA-Z0-9_]*$/;
        if (regex.test(userName) == false) {
            $("#ver_UserNameMsg").html("<font color='red'>只能以英文字母或下划线开头,只能包含英文字母，下划线或数字</font>");
            return false;
        } else {
            $("#ver_UserNameMsg").html("<font color='green'>√</font>");
            flagUser = true
        }
    }

    function checkPassword() {
        var password = $("#Password").val();
        var regex = /^(?![0-9]+$)(?![a-z]+$)(?![A-Z]+$).{6,}$/;

        if (regex.test(password) == false) {
            $("#ver_PasswordMsg").html("<font color='red'>长度在 6 位以上，不能含有 3 位以上的连续数字，必须有大写字母，小写字母或数字中的两项 </font>");
        } else {
            $("#ver_PasswordMsg").html("<font color='green'>√</font>");
            flagPwd = true
        }
    }

    function checkRepeatPassword() {
        var password = $("#Password").val();
        var RepeatPassword = $("#RepeatPassword").val();

        if (password == RepeatPassword && RepeatPassword != "" && RepeatPassword != null && RepeatPassword != undefined) {
            $("#ver_RepeatPasswordMsg").html("<font color='green'>√</font>");
            flagRpwd = true
        } else {
            $("#ver_RepeatPasswordMsg").html("<font color='red'>两次密码不一致或密码为空</font>");
        }
    }

    $(".submit").on('click', function (e) {
        if (flagUser && flagPwd && flagRpwd) {
            var userName = $("#UserName").val();
            var password = $("#Password").val();
            var RepeatPassword = $("#RepeatPassword").val();
            $.post('/index/register',
                {
                    userName: userName,
                    password: password,
                    repeatPassword: RepeatPassword
                },
                function (res) {
                    res = JSON.parse(res)
                    console.log(res)
                    if (res.code == 200) {
                        alert(res.msg);
                    }
                }
            )
        } else {
            checkUserName()
            checkPassword()
            checkRepeatPassword()
        }
    })
</script>
