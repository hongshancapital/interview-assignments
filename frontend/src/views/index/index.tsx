import React from 'react';
import Carousel from '../../components/carousel'
import './index.scss';

const Index = () => {

    return (
        <Carousel duration={2}>
            <div className="content content1">
                <h1>xPhone</h1>
                <p>Lots to love.Less to spend.</p>
                <p>Starting at $399</p>
            </div>
            <div className="content content2">
                <h1>Tablet</h1>
                <p>Just the right amount of everything.</p>
            </div>
            <div className="content content3">
                <h1>Buy a Tablet or xPhone for college.</h1>
                <h1>Get arPods.</h1>
            </div>
        </Carousel>
    );
}

export default Index;