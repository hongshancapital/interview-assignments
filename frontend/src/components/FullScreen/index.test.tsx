import { render } from '@testing-library/react';
import React from 'react';
import { FullScreen } from './index';

test('render full screen component', () => {
  const { getByTestId, container, rerender } = render(
    <FullScreen>
      <div data-testid="child">Hello World</div>
    </FullScreen>
  );
  expect(getByTestId('child')).toBeInTheDocument();
  expect(getByTestId('child').textContent).toBe('Hello World');
  const compDom = container.firstChild as HTMLElement;
  expect(compDom).not.toBeNull();
  expect(compDom).toHaveClass('full');
  expect(compDom).toHaveStyle(`width: ${window.innerWidth}px; height: ${window.innerHeight}px`);
  rerender(<FullScreen center>Hello</FullScreen>);
  expect(compDom).toHaveClass('center');
  rerender(<FullScreen className="bbb">Hello</FullScreen>);
  expect(compDom).toHaveClass('bbb');
  rerender(<FullScreen style={{ backgroundColor: 'red' }}>Hello</FullScreen>);
  expect(compDom).toHaveStyle('background-color: red');
  expect(container).toMatchSnapshot();
});
