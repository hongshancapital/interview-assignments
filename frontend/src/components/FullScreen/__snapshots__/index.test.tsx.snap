// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render full screen component 1`] = `
<div>
  <div
    class="full"
    style="width: 1024px; height: 768px; background-color: red;"
  >
    Hello
  </div>
</div>
`;
